### PR TITLE
Fix arena locking in tcache_bin_flush_large().

### PR DESCRIPTION
**Pull request by interwq**
This regression was introduced in c834912 (incorrect arena used).

### CODE REVIEW

File: src/tcache.c [hunk 1]
```diff
@@ -212,7 +212,7 @@ tcache_bin_flush_large(tsd_t *tsd, cache_bin_t *tbin, szind_t binind,
 			idump = false;
 		}
 
-		bool lock_large = !arena_is_auto(arena);
```
Comment (interwq) [inline src/tcache.c]: This was the bug.

**Comment by davidtgoldblatt** [comment]
Is there a test case we could add that would catch this?

**Comment by interwq** [comment]
re: test case. Thought about this a bit, but leaning to say this should be taken care of by a stress test instead. We have some assertions for cases like this, e.g. https://github.com/jemalloc/jemalloc/blob/c834912aa9503d470c3dae2b2b7840607f0d6e34/src/large.c#L334
however in this instance we'd need to do a producer-consumer. I'll try adding that pattern somewhere.

**Comment by interwq** [comment]
Just added a new test case for remote free + tcache_flush. Verified that it catches the issue being fixed here.

File: test/integration/mallocx.c [hunk 2]
```diff
@@ -71,6 +71,38 @@ TEST_BEGIN(test_overflow) {
...
+	    "Unexpected mallctl() failure");
+	size_t large_sz;
+	sz = sizeof(size_t);
+	assert_d_eq(mallctl("arenas.lextent.0.size", (void *)&large_sz, &sz,
+	    NULL, 0), 0, "Unexpected mallctl failure");
+
+	void *ptr = mallocx(large_sz, MALLOCX_ARENA(arena)
+	    | MALLOCX_TCACHE_NONE);
+	void **ret = (void **)arg;
+	*ret = ptr;
```
Comment (interwq) [inline test/integration/mallocx.c]: Had to pass back the address this way, since the thread exit code on Windows is 32bit.

**Comment by interwq** [comment]
cc @davidtgoldblatt in case you want to take a look at the test case.

**Pull request merged**